gr-filter/lib/freq_xlating_fir_filter_impl.cc

```diff
@@ -175,10 +176,12 @@ int freq_xlating_fir_filter_impl<IN_T, OUT_T, TAP_T>::work(
 
     unsigned j = 0;
     for (int i = 0; i < noutput_items; i++) {
-        out[i] = d_r.rotate(d_composite_fir->filter(&in[j]));
-        j += this->decimation();
+        out[i] = d_composite_fir->filter(&in[j]);
```

Comment:
While using `out` as a temporary buffer will work, I'd prefer to use an actual temporary buffer instead, just to be memory-safe. So, using a `std::vector` of the correct type and size here for the output and then as the input for line 183. I know this change isn't technically necessary since the rotator -should- be able to handle in-place computations ... but, IMHO better safe than sorry! If anyone disagrees, please talk me out of it...

Reply:
With how frequently work gets called, newing and deleting memory for a temp buffer can be expensive.  You can do it with a class-level buffer that gets allocated once and destroyed in stop/destructor and only re-allocated if the necessary size is greater than the currently allocated size.  But there's a number of volk kernels where it is safe to use the same space in in and out.  In the interest of performance if it works okay and is safe this way, i wouldn't complicate the class by doing additional temp buffer handling.  Just my personal preference.

Reply:
Ok. Then let me change my request: add a comment here to the effect of what you just wrote, about how `out` is being used then reused here and why, how it interfaces with Volk, etc...

Reply:
Happy to add comments.  Were you thinking in association with the "out[i] = d_r.rotate(d_composite_fir->filter(&in[j]))" line or the "d_r.rotateN(out, out, noutput_items)" line?  I think it would make more sense on the 2nd since the first really wouldn't require any temp space since out isn't on the right side of the = anywhere?

Reply:
Since the code is so short, I'd say before the line `unsigned j = 0;` ... anyone reading the comment can quickly the following lines to understand what's going on then. And, doesn't interrupt code flow. Thx!